File: Lista01/batalhaNaval/App4.java
package Lista01.batalhaNaval;
import java.util.Scanner;
import java.util.Random;

// Jodo da batalha naval
public class App4 {
    static final short QTD_NAVIOS = 5;
    static final int MAX_RANDOM = 7;
    static Jogador player1 = new Jogador(1);
    static Jogador player2 = new Jogador(2);
    public static void main(String[] args) {
        intro();
        //variaveis e objetos
        Random rand = new Random();
        String msg;
        int tempx, tempy;
        boolean atirou;

        // pergunta as coordenadas
        System.out.println(player1.nome + ", enter your ships' coordinates.");
        for (short x=1; x<=QTD_NAVIOS; ){
            System.out.println("Enter ship " + x + " location > x y <:");
            tempx= rand.nextInt(MAX_RANDOM);
            tempy= rand.nextInt(MAX_RANDOM);
            System.out.println(tempx + " "+ tempy);
            msg = player1.insereCoordenadas(tempx, tempy);
            if (msg != "") System.out.println(msg);
            else x++;
        }
        player1.exibeMapaPlayer();
        aguarde(3);
        limpaTela();

        // pergunta as coordenadas
        System.out.println(player2.nome + ", enter your ships' coordinates.");
        for (short x=1; x<=QTD_NAVIOS; ){
            System.out.println("Enter ship " + x + " location > x y <:");
            tempx= rand.nextInt(MAX_RANDOM);
            tempy= rand.nextInt(MAX_RANDOM);
            System.out.println(tempx + " "+ tempy);
            msg = player2.insereCoordenadas(tempx, tempy);
            if (msg != "") System.out.println(msg);
            else x++;
        }
        player2.exibeMapaPlayer();
        aguarde(3);
        limpaTela();
        
        while(true){
            // player 1 atira
            do{
                System.out.println(player1.nome + ", enter hit row/column:");
                tempx= rand.nextInt(MAX_RANDOM);
                tempy= rand.nextInt(MAX_RANDOM);
                System.out.println(tempx + " "+ tempy);

                atirou = player1.atiraEm(player2, tempx, tempy);
            } while(!atirou);
            // exibe o mapa do player 2
            player2.exibeMapaGame();
            if(player1.ganhou) {
                fimGame(player1.nome);
                break;
            }
            
            // player 2 atira
            do{
                System.out.println(player2.nome + ", enter hit row/column:");
                tempx= rand.nextInt(MAX_RANDOM);
                tempy= rand.nextInt(MAX_RANDOM);
                System.out.println(tempx + " "+ tempy);
                
                atirou = player2.atiraEm(player1, tempx, tempy);
            } while(!atirou);
            // exibe o mapa do player 1
            player1.exibeMapaGame();
            if(player2.ganhou) {
                fimGame(player2.nome);
                break;
            }
        }
    }

    public static void intro(){
        limpaTela();
        System.out.println("\tWelcome to Battleship!!!");
        System.out.println("\n\n\n");
        aguarde(2);
        limpaTela();
    }

    public static void fimGame(String ganhador){
        System.out.println(ganhador + " WINS! You sunk all of your opponent's ships!");
        System.out.println("Map of "+ player1.nome);
        player1.exibeMapaPlayer();
        System.out.println("Map of "+player2.nome);
        player2.exibeMapaPlayer();
    }

    public static void limpaTela(){
        for (short i = 1; i<100; i++) System.out.println("\n");
    }

    public static void aguarde(int segundos){
        try {
            Thread.sleep(segundos*1000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}

File: Lista02/batalhaNaval/app4.py
from jogador import Jogador
import random, time, os

# variaveis
MAX_RANDOM = 7
player1 = Jogador(1);
player2 = Jogador(2);
numerosRandomicos = True
# se true, o jogo roda com números aleatórios
# se false, o jogo roda com entradas do usuario

# funcoes
def limpaTela():
    retorno = os.system('clear')
    if retorno != 0:
        os.system('clear')

def recebeValores():
    if numerosRandomicos:
        return random.randint(0, MAX_RANDOM+1), random.randint(0, MAX_RANDOM+1)
    x, y = input().split(' ')
    x = int(x)
    y = int(y)
    return x, y

def recebeCoordenada():
    tempx, tempy = recebeValores()
    if numerosRandomicos:
        print(f'{tempx} {tempy}')
    return tempx, tempy

def intro():
    limpaTela()
    print("\n\n\n\tWelcome to Battleship!!!")
    print('\n')
    delay()
    limpaTela()

def fimGame(ganhador: Jogador):
    limpaTela()
    print(f'{ganhador.getNome()} WINS! You sunk all of your opponent\'s ships!')
    print(f'Map of {player1.getNome()}')
    player1.exibeMapaPlayer()
    print(f'Map of {player2.getNome()}')
    player2.exibeMapaPlayer()

def delay(segundos=2):
    time.sleep(segundos)

def pegaCoordenadas(jogador:Jogador):
    print(f'{jogador.getNome()}, enter yours ship\'s coordinates.')
    i = 0
    while i < Jogador.qtdNavios:
        print(f'Enter ship {i+1} location > x y <:')
        tempx, tempy = recebeCoordenada()
        msg = jogador.insereCoordenadas(tempx, tempy)
        if msg!='': print(msg)
        else: i+=1
    jogador.exibeMapaPlayer()
    delay(3)
    limpaTela()

def tiro(atirador:Jogador, alvo:Jogador):
    while True:
        print(f'{atirador.getNome()}, enter hit row/column:')
        tempx, tempy = recebeCoordenada()
        atirou = atirador.atiraEm(alvo, tempx, tempy)

        if atirou:
            break
    
    alvo.exibeMapaGame()

def Ganhador(jogador: Jogador):
    if jogador.ganhou():
        fimGame(jogador)
        return True
    return False

# inicio
def main():
    intro()
    pegaCoordenadas(player1)
    pegaCoordenadas(player2)

    while True:
        tiro(player1, player2)
        if numerosRandomicos: delay()
        if Ganhador(player1): break

        tiro(player2, player1)
        if numerosRandomicos: delay()
        if Ganhador(player2): break

if __name__ == '__main__':
    main()